compiler/parser/RdrHsSyn.hs

```diff
@@ -2848,6 +2872,78 @@ mkArchiveChoice =
   where
     pureUnit = mkApp (mkUnqualVar $ mkVarOcc "pure") (noLoc $ ExplicitTuple noExt [] Boxed)
 
+validateException
+  :: Located RdrName
+  -> ExceptionBodyDecls
+  -> P ValidException
+validateException name ExceptionBodyDecls{..}
+  | length ebdMessage > 1 = report "Multiple 'message' declarations"
+  | otherwise =
+      pure ValidException
+        { veName = name
+        , veMessage = listToMaybe ebdMessage
+        }
+  where
+    report :: String -> P a
+    report e = addFatalError (getLoc name) (text e)
```

Comment:
can we throw diagnostics on the message instead of the exception type? maybe even for all except for the first.

Reply:
That's a good idea :)